cmd/healthchecker/options/options.go

```diff
@@ -0,0 +1,76 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"flag"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	"k8s.io/node-problem-detector/pkg/healthchecker/types"
+)
+
+// NewHealthCheckerOptions returns an empty health check options struct.
+func NewHealthCheckerOptions() *HealthCheckerOptions {
+	return &HealthCheckerOptions{}
+}
+
+// HealthCheckerOptions are the options used to configure the health checker.
+type HealthCheckerOptions struct {
+	Component          string
+	ContainerRuntime   string
+	EnableRepair       bool
+	CriCtlPath         string
+	CoolDownTime       time.Duration
+	HealthCheckTimeout time.Duration
+}
+
+// AddFlags adds health checker command line options to pflag.
+func (hco *HealthCheckerOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&hco.Component, "component", types.KubeletComponent,
+		"The component to check health for. Supports kubelet and container-runtime. Defaults to kubelet.")
+	fs.StringVar(&hco.ContainerRuntime, "container-runtime", types.DockerRuntime,
+		"The underlying container runtime name on the node. This is ignored if component is not container-runtime. Supports docker and containerd. Defaults to docker.")
+	fs.BoolVar(&hco.EnableRepair, "enable-repair", true, "Flag to enable/disable repair attempt for the component. Defaults to true.")
+	fs.StringVar(&hco.CriCtlPath, "crictl-path", types.DefaultCriCtl,
+		"The path to the crictl binary. This is used to check health of container-runtimes other than docker. Defaults to /usr/bin/crictl")
+	fs.DurationVar(&hco.CoolDownTime, "cooldown-time", types.DefaultCooldownTime,
+		"The time to wait after repairing the component. This helps in maintaining a buffer between repair and next invocation. Defaults to 2m.")
+	fs.DurationVar(&hco.HealthCheckTimeout, "health-check-timeout", types.DefaultHealthCheckTimeout,
+		"The time to wait before marking the component as unhealthy. Defaults to 10s.")
+}
+
+// ValidOrDie validates health checker command line options.
+func (hco *HealthCheckerOptions) ValidOrDie() {
+	// Make sure the component specified is valid.
+	if hco.Component != types.KubeletComponent && hco.Component != types.ContainerRuntimeComponent {
+		panic("The component specified is not supported. Supported components are : <docker/container-runtime>")
+	}
+	// Skip checking for container-runtime if the component is not container-runtime
+	if hco.Component != types.ContainerRuntimeComponent {
+		return
+	}
+	// Make sure the container-runtime is valid.
+	if hco.ContainerRuntime != types.DockerRuntime && hco.ContainerRuntime != types.ContainerdRuntime {
+		panic("The container-runtime specified is not supported. Supported runtimes are : <docker/containerd>")
+	}
+}
```

Comment:
The path has a default value which is used in case it is empty.

Reply:
No, it will use the empty string. I have added a test to detect that. But in general, if that flag is set there is no way of making sure that path exists in the test.